### DISCUSSION TITLE
Argument of type 'Root' is not assignable to parameter of type 'Node<Data>'

### DISCUSSION THREAD

**Discussion by dmca-glasgow** [syntax-tree/unist#129] [Q&A] [answered]
Hi there 👋

I have been updating my dependencies and now `tsc` is complaining about `unist-util-visit`.

Given the following:

```ts
import { Image, Root } from 'mdast';
import { visit } from 'unist-util-visit';

export function inlineImages() {
  return async (tree: Root) => {
    const nodes: Image[] = [];

    visit(tree, 'image', (node) => {
      nodes.push(node);
    });

    // ...
  };
}
```

It seems I'm now not allowed to pass a `Root` type as 1st argument to `visit`:

![Screenshot 2024-07-24 at 14 59 19](https://github.com/user-attachments/assets/3bd525e5-f28f-4ac5-a177-9010c7b51fa1)

However I'm unsure how to type the suggested `Node<Data>`.  I've tried `Node` from `mdast` and `unist` but to no avail.

Any help greatly appreciated 😎

**Comment by ChristianMurphy** [comment, marked as answer]
Hey @dmca-glasgow 👋 
That error looks a lot like you may be mixing major versions of `unist-util-visit` and `@types/mdast`.
Make sure are both on the latest major version, at the time of writing these are `@types/mdast` v4 and `unist-util-visit` v5.

https://stackblitz.com/edit/github-4yca6z?file=src%2Fmain.ts

**Reply by dmca-glasgow** [reply]
Many thanks @ChristianMurphy
